src/libs/Navigation/Navigation.ts

```diff
@@ -307,57 +284,53 @@ function setIsNavigationReady() {
 /**
  * Checks if the navigation state contains routes that are protected (over the auth wall).
  *
- * @function
- * @param {Object} state - react-navigation state object
- *
- * @returns {Boolean}
+ * @param state - react-navigation state object
  */
-function navContainsProtectedRoutes(state) {
-    if (!state || !state.routeNames || !_.isArray(state.routeNames)) {
+function navContainsProtectedRoutes(state: NavigationState | undefined): boolean {
+    if (!state?.routeNames || !Array.isArray(state.routeNames)) {
         return false;
     }
 
-    const protectedScreensName = _.values(PROTECTED_SCREENS);
-    const difference = _.difference(protectedScreensName, state.routeNames);
+    const protectedScreensName = Object.values(PROTECTED_SCREENS);
+    const difference = protectedScreensName.filter((screen) => !state.routeNames.includes(screen));
 
     return !difference.length;
 }
 
 /**
- * Waits for the navitgation state to contain protected routes specified in PROTECTED_SCREENS constant.
- * If the navigation is in a state, where protected routes are avilable, the promise resolve immediately.
+ * Waits for the navigation state to contain protected routes specified in PROTECTED_SCREENS constant.
+ * If the navigation is in a state, where protected routes are available, the promise resolve immediately.
  *
  * @function
- * @returns {Promise<void>} A promise that resolves when the one of the PROTECTED_SCREENS screen is available in the nav tree.
+ * @returns A promise that resolves when the one of the PROTECTED_SCREENS screen is available in the nav tree.
  *
  * @example
  * waitForProtectedRoutes()
  *     .then(()=> console.log('Protected routes are present!'))
  */
 function waitForProtectedRoutes() {
-    return new Promise((resolve) => {
+    return new Promise<void>((resolve) => {
         isNavigationReady().then(() => {
-            const currentState = navigationRef.current.getState();
+            const currentState = navigationRef.current?.getState();
             if (navContainsProtectedRoutes(currentState)) {
                 resolve();
                 return;
             }
-            let unsubscribe;
-            const handleStateChange = ({data}) => {
-                const state = lodashGet(data, 'state');
+            let unsubscribe: (() => void) | undefined;
```

Comment:
Could we simplify this by declaring the variable in the same statement and using inferred types?